experimental/dds/tree2/src/test/shared-tree/editing.spec.ts

```diff
@@ -1781,15 +1854,79 @@ describe("Editing", () => {
 	});
 
 	describe("Optional Field", () => {
-		// TODO unskip these tests once optional fields track transient nodes
-		it.skip("can rebase a node replacement and a dependent edit to the new node", () => {
+		describe("can rebase a set over another set", () => {
+			it("from a non-empty state", () => {
+				const tree1 = makeTreeFromJson([{ foo: "1" }]);
+				const tree2 = tree1.fork();
+				const tree3 = tree1.fork();
+
+				tree2.editor
+					.valueField({ parent: rootNode, field: brand("foo") })
+					.set(cursorForJsonableTreeNode({ type: leaf.string.name, value: "2" }));
+
+				tree3.editor
+					.valueField({ parent: rootNode, field: brand("foo") })
+					.set(cursorForJsonableTreeNode({ type: leaf.string.name, value: "3" }));
+
+				tree3.rebaseOnto(tree2);
+				tree2.merge(tree3, false);
+				tree1.merge(tree3, false);
```

Comment:
Seems correct to me, but hard to follow. May I interest you in a different approach?
```suggestion
				tree1.merge(tree2, false);
				tree1.merge(tree3, false);
				tree2.rebaseOnto(tree1);
				tree3.rebaseOnto(tree1);
```
The first two lines define the edit sequencing, while the last two just represent the participants learning of that ordering.